exp/zapslog/go.mod

```diff
@@ -0,0 +1,13 @@
+module go.uber.org/zap/exp/zapslog
```

Comment:
I wasn't planning on the subpackage being a submodule, but I'm not opposed to it.
@mway @prashantv @sywhang what do you think?

- Pro: Independently versioned. Can tag `exp/zapslog/v0.1.0` (or `exp/v0.1.0` if we move the submodule definition one level up) independently of Zap's 1.0 instead of saying "go.uber.org/zap/exp is part of the main Zap module, but it doesn't comply with semver."
- Con: Adds slightly more complexity to the repository. Have to use replace directives or go.work files for cross-cutting changes (because exp will otherwise only have access to the released version), and have to be sure to release Zap before exp if a change in exp depends on a new change in Zap.

Reply:
More modules does seem more complex, though separating exp (where we probably will keep it at 0.x and make breaking changes) seems like a good idea. So submodule at exp is my vote.

Reply:
Per the discussion, I have made the `exp` a submodule.

Reply:
Excellent, thanks!